c/zss.c

```diff
@@ -698,37 +698,81 @@ static void installWebPluginDefintionsService(WebPluginListElt *webPlugins, Http
   zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG2, "end %s\n", __FUNCTION__);
 }
 
-static int checkLoggingVerbosity(const char *serverConfigFile, char *pluginIdentifier, ShortLivedHeap *slh) {
-  char errorBuffer[1024] = {0};
-  
-  Json *json = jsonParseFile(slh, serverConfigFile, errorBuffer, sizeof(errorBuffer));
-  if (json) {
-    if (jsonIsObject(json)) {
-      JsonObject *jsonObject = jsonAsObject(json);
-      JsonObject *logLevels = jsonObjectGetObject(jsonObject, "logLevels");
-      if (logLevels != NULL) {
-        JsonProperty *property = jsonObjectGetFirstProperty(logLevels);
-        while (property != NULL) {
-          if (!strcmp(jsonPropertyGetKey(property), pluginIdentifier)) {
-            int logLevel = jsonObjectGetNumber(logLevels, pluginIdentifier);
-            if (logLevel <= ZOWE_LOG_DEBUG3 || logLevel >= ZOWE_LOG_ALWAYS) {
-              return logLevel;
-            }
-            else {
-              break;
-            }
-          }
-          property = jsonObjectGetNextProperty(property);
+#define ENV_LOGLEVELS_PREFIX "ZWED_logLevels_"
+static int getLogLevelsFromEnv(JsonObject *envConfig, const char *identifier) {
+  int len = strlen(ENV_LOGLEVELS_PREFIX) + strlen(identifier) + 1;
+  int logLevel = ZOWE_LOG_SEVERE;
+  char envKey[len];
+  memset(envKey,'\0',len);
+  snprintf(envKey, len, "%s%s", ENV_LOGLEVELS_PREFIX, identifier);
+  logLevel = jsonObjectGetNumber(envConfig, envKey);
+  return logLevel;
```

Comment:
Again, the function returns `0` if the environment variable for the `logLevel`
* is not set
* is set to `0`

We should distinguish the cases.

Reply:
Added jsonObjectHasKey() check before jsonObjectGetNumber() call.